ibm_cloud_sdk_core/base_service.py

```diff
@@ -240,11 +266,18 @@ def prepare_request(self,
                 or a list of tuples [ (<part-name>, (<tuple>))... ]
 
         Raises:
-            ValueError: If service_url is not specified.
+            ValueError: If service_url is not specified, or authenticator is invalid.
 
         Returns:
             Prepared request dictionary.
         """
+        if not self.authenticator:
```

Comment:
I apologize for going back and forth on this, but the validation of the existence of the authenticator should be left in the BaseService ctor.   As we discussed in our quick start session today, my recollection is that we previously agreed (during the design of the service factory feature) to defer that until first use, but Mike remembered differently :)

Reply:
👍